integration-tests/tracecontext/src/main/java/io/opentelemetry/Application.java

```diff
@@ -34,99 +41,93 @@
-                          if (headers == null || !headers.hasMoreElements()) {
-                            return null;
-                          }
-                          List<String> values = new ArrayList<>();
-                          while (headers.hasMoreElements()) {
-                            String nextElement = headers.nextElement();
-                            if (!nextElement.trim().isEmpty()) {
-                              values.add(nextElement);
-                            }
-                          }
-                          if (values.isEmpty()) {
-                            return null;
-                          }
-                          if (values.size() == 1) {
-                            return values.get(0);
-                          }
-                          StringBuilder builder = new StringBuilder(values.get(0));
-                          for (int i = 1; i < values.size(); i++) {
-                            builder.append(",").append(values.get(i));
-                          }
-
-                          return builder.toString();
-                        }
-                      });
-
-          for (io.opentelemetry.Request req : requests) {
-            Span span =
-                openTelemetry
-                    .getTracer("validation-server")
-                    .spanBuilder("Entering Validation Server")
-                    .setParent(context)
-                    .startSpan();
-
-            Context withSpanContext = context.with(span);
-
-            // Make a new request using the builder
-            okhttp3.Request.Builder reqBuilder = new okhttp3.Request.Builder();
-
-            // Inject the current context into the new request.
-            openTelemetry
-                .getPropagators()
-                .getTextMapPropagator()
-                .inject(withSpanContext, reqBuilder, okhttp3.Request.Builder::addHeader);
-
-            // Add the post body and build the request
-            String argumentsJson = gson.toJson(req.getArguments());
-            RequestBody argumentsBody =
-                RequestBody.create(
-                    argumentsJson, MediaType.parse("application/json; charset=utf-8"));
-            okhttp3.Request newRequest = reqBuilder.url(req.getUrl()).post(argumentsBody).build();
-
-            // Execute the request
-            OkHttpClient client = new OkHttpClient();
-            try (okhttp3.Response res = client.newCall(newRequest).execute()) {
-              logger.info("response: " + res.code());
-            } catch (Exception e) {
-              logger.log(Level.SEVERE, "failed to send", e);
-            }
-            span.end();
-          }
-
-          return "Done";
-        });
-
-    Spark.awaitInitialization();
+    WebClient client = WebClient.of();
+
+    Server server =
+        Server.builder()
+            .http(5000)
+            .annotatedService(
+                new Object() {
```

Comment:
I believe that splitting this into a separate class (even private) would improve readability a bit.

Reply:
Sure